### PR TITLE
Feature/process data on prod

### LINKED ISSUE #64: add a github action to publish datasets(process_data) to production
## Use Case

_Please provide a use case to help us understand your request in context_

We want data creators to be able to test their data also in production database before releasing it

## Acceptance Criteria

_Please describe how you know this is done_

An authorized user should be able to run `process-dataset` on `main` to publish their dataset to production database. The job will be skipped if it's on other branches.   

## Details

_Please provide any helpful specifications_

### Plan with manually triggered github action:  

1. Run `process_dataset` to publish a new dataset to staging db (github action no.1) --DONE
2. Test and check data at staging frontend 
3. Run `process_dataset` to publish to production db (github action no.2) --TODO, this is the new issue #64
4. Test and check data at both stag & prod frontend 
5. After testing, release dataset to production database (github action no.3) --ALMOST DONE

### PR DESCRIPTION
**Pull request by rugeli**
Problem
=======
Closes #64 

Solution
========
What I/we did to solve this problem

- added a choice input for users to decide which db (stag or prod) they want to publish dataset to
- In `upload-dataset.yml`, added a second job that calls github api to get collaborators with admin access, and outputs all the admins for the next job to use 
- added a third job that will run `npm run process-dataset` with production DB. Only on `main` by admins

Now we have 2 manually triggered files in our workflows: 
1. `release-dataset.yml`: release data to staging or production, the job will only be triggered on `main` by authorized users for both databases
2. `upload-dataset.yml`: upload data to staging or production, anyone with write access to this repo should be able to publish data from any branches to **staging**, but only authorized users can publish data only from `main` branch to **production**     

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)

* Repo admins can manage access [here](https://github.com/allen-cell-animated/cell-feature-data/settings/access)

## Steps to Verify
1. Go to Actions and choose `Upload dataset to staging or production DB` in workflows
2. Click on Run Workflow and select `feature/process-data-on-prod` branch for testing. 
    - Note that the job `process_dataset_to_production` is expected to be skipped on a non-main branch, comment out `if:  github.ref == 'refs/heads/main'` to run the job
4. Enter inputs and comment out `npm run process-dataset` for production DB testing
5. Run workflow

## Expected behavior 
- when DB == 'staging'
Screenshot 2023-04-03 at 12 36 44 PM

- when DB == 'production' && authorized user && on `main`
Screenshot 2023-04-03 at 12 36 58 PM

- in the prod job, if not an admin, step`process dataset to production` will be skipped, step `Not authorized to process dataset to production` will run the echo statement 
Screenshot 2023-04-03 at 12 38 22 PM

### CODE REVIEW

File: .github/workflows/stage-dataset.yml [hunk 1]
```diff
@@ -10,23 +10,54 @@ on:
         type: boolean
         description: "Set to true to skip the file-info upload (save some processing time)"
```
Comment (meganrm) [inline .github/workflows/stage-dataset.yml:11]: suggestion for clarity: "Skip the file-info upload to save some processing time. Only select if this isn't the first time uploading the dataset, and there are no changes to per cell file-info"

File: .github/workflows/stage-dataset.yml [hunk 2]
```diff
@@ -10,23 +10,54 @@ on:
...
         default: false
+      database:
+        description: "Please choose a database to release dataset"
```
Comment (toloudis) [inline .github/workflows/stage-dataset.yml:14]: I think we should change the word `release` here, as we are not releasing anything.  Maybe `upload` is a better word choice?

**Comment by toloudis** [comment]
Who owns the list of authorized users?  Is it a different list for uploading production data vs releasing it?

**Comment by rugeli** [comment]
We have a github repo secret `OWNERS`, which contains a list of authorized usernames. I believe all three of us have the right to update the secret. Uploading and releasing production data share the same secret owner list.

**Comment by rugeli** [comment]
Note that as we changed the file's name from `stage-dataset.yml` to `upload-dataset.yml`, we will see `Workflow does not exist or does not have a workflow_dispatch trigger in this branch.` when we attempt to test the workflow on our testing branch. Need to merge the change to main to update the action name

**Review by meganrm — APPROVED** [review]

**Review by toloudis — APPROVED** [review]

**Pull request merged**